src/components/mode-tools/mode-tools.jsx

```diff
@@ -33,185 +37,446 @@ import straightPointIcon from './icons/straight-point.svg';
-            id: 'paint.modeTools.flipHorizontal'
-        },
-        flipVertical: {
-            defaultMessage: 'Flip Vertical',
-            description: 'Label for the button to flip the image vertically',
-            id: 'paint.modeTools.flipVertical'
+const messages = defineMessages({
+    brushSize: {
+        defaultMessage: 'Brush size',
+        description: 'Label for the brush size input',
+        id: 'paint.modeTools.brushSize'
+    },
+    lineSize: {
+        defaultMessage: 'Line size',
+        description: 'Label for the line size input',
+        id: 'paint.modeTools.lineSize'
+    },
+    eraserSize: {
+        defaultMessage: 'Eraser size',
+        description: 'Label for the eraser size input',
+        id: 'paint.modeTools.eraserSize'
+    },
+    copy: {
+        defaultMessage: 'Copy',
+        description: 'Label for the copy button',
+        id: 'paint.modeTools.copy'
+    },
+    paste: {
+        defaultMessage: 'Paste',
+        description: 'Label for the paste button',
+        id: 'paint.modeTools.paste'
+    },
+    curved: {
+        defaultMessage: 'Curved',
+        description: 'Label for the button that converts selected points to curves',
+        id: 'paint.modeTools.curved'
+    },
+    pointed: {
+        defaultMessage: 'Pointed',
+        description: 'Label for the button that converts selected points to sharp points',
+        id: 'paint.modeTools.pointed'
+    },
+    flipHorizontal: {
+        defaultMessage: 'Flip Horizontal',
+        description: 'Label for the button to flip the image horizontally',
+        id: 'paint.modeTools.flipHorizontal'
+    },
+    flipVertical: {
+        defaultMessage: 'Flip Vertical',
+        description: 'Label for the button to flip the image vertically',
+        id: 'paint.modeTools.flipVertical'
+    },
+    sansSerif: {
+        defaultMessage: 'Sans Serif',
+        description: 'Name of the sans serif font',
+        id: 'paint.modeTools.sansSerif'
+    },
+    serif: {
+        defaultMessage: 'Serif',
+        description: 'Name of the serif font',
+        id: 'paint.modeTools.serif'
+    },
+    handwriting: {
+        defaultMessage: 'Handwriting',
+        description: 'Name of the handwriting font',
+        id: 'paint.modeTools.handwriting'
+    },
+    marker: {
+        defaultMessage: 'Marker',
+        description: 'Name of the marker font',
+        id: 'paint.modeTools.marker'
+    },
+    curly: {
+        defaultMessage: 'Curly',
+        description: 'Name of the curly font',
+        id: 'paint.modeTools.curly'
+    },
+    pixel: {
+        defaultMessage: 'Pixel',
+        description: 'Name of the pixelated font',
+        id: 'paint.modeTools.pixel'
+    }
+});
+class ModeToolsComponent extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'getFontStyle',
+            'getTranslatedFontName',
```

Comment:
Since all these methods are used only for the font dropdown, would you mind resetting this to a functional component and create a separate container/component for doing the event handling for the font dropdown? It would help keep this file a bit more easy to read, and keep the separation between containers/components.

Reply:
Done